github_issue_summarization/serving_the_model.md

```diff
@@ -1,21 +1,62 @@
 # Serving the model
 
-We are going to use a simple tornado server to serve the model. The [server.py](notebooks/server.py) contains the server code.
+We are going to use [seldon-core](https://github.com/SeldonIO/seldon-core) to serve the model. [IssueSummatization.py](notebooks/IssueSummatization.py) contains the code for this model. We will wrap this class into a seldon-core microservice which we can then deploy as a REST or GRPC API server.
 
-Start the server using `python server.py --port=8888`.
+> The model is written in Keras and when exported as a TensorFlow model seems to be incompatible with TensorFlow Serving. So we're using seldon-core to serve this model since seldon-core allows you to serve any arbitrary model. More details [here](https://github.com/kubeflow/examples/issues/11#issuecomment-371005885).
 
-> The model is written in Keras and when exported as a TensorFlow model seems to be incompatible with TensorFlow Serving. So we're using our own webserver to serve this model. More details [here](https://github.com/kubeflow/examples/issues/11#issuecomment-371005885).
+# Prerequisites
 
-## Sample request
+Ensure that you have the following files from the [training](training_the_model.md) step in your `notebooks` directory:
+
+* `seq2seq_model_tutorial.h5` - the keras model
+* `body_pp.dpkl` - the serialized body preprocessor
+* `title_pp.dpkl` - the serialized title preprocessor
+
+# Wrap the model into a seldon-core microservice
+
+cd into the notebooks directory and run the following docker command. This will create a build/ directory.
+
+```
+cd notebooks/
```

Comment:
Does Seldon's model wrapper support reading the model from GCS/S3?

Could we run the python wrapper as a job on K8s? Could we create a ksonnet template to make it easy to launch such a job.

I don't think we should try to address either of those issues in this PR. But might be good to open up issues to track them.

/cc @cliveseldon @gsunner @ahousley

Reply:
> Does Seldon's model wrapper support reading the model from GCS/S3?

Not that I am aware of. Looking at the [instructions](https://github.com/SeldonIO/seldon-core/blob/master/docs/wrappers/python.md#create-a-model-folder), it seems to require all the files in a single folder which contains the code and the model to build the serving image.

Created an issue to track the feature for serving a seldon-core model: https://github.com/kubeflow/kubeflow/issues/405

Reply:
Looks like there is a seldon [component](https://github.com/kubeflow/kubeflow/tree/master/kubeflow/seldon) in kubeflow. In the next PR, I'll try to deploy this image on k8s using this component

Reply:
You can specify any persistent volume in definition of the SeldonDeployment in the componentSpec as its a standard Kubernetes PodTemplateSpec. So you can load your parameters from their rather than package them as part of the model image if you wish. The [kubeflow-seldon example](https://github.com/kubeflow/example-seldon) shows an example using an NFS volume to load the trained model parameters.